CONTRIBUTING.md

```diff
@@ -7,20 +7,20 @@ You should first understand what the hell you're doing.
 
 # Setup repo
 ```
-# Public fork of Apache Spark.  Used for public pull requests.
-git remote add origin git@github.com:mesosphere/spark.git
-git fetch origin
+# Set up Mesosphere Spark, a public fork of Apache Spark, used for public pull requests.
+git clone git@github.com:mesosphere/spark.git
 
-# Apache Spark
+# Add apache spark as upstream
 git remote add upstream git@github.com:apache/spark
-git fetch upstream
+# Depending of if you want to continue work on upstream master/branch
+git pull  git pull upstream master/branch
```

Comment:
Fixing the multiple pull typo, first up.
So while the previous code did fetch upstream changes for all branches, to get relevant changes on the specific branch we still have to merge that branch upstream to current. And yes agree it might lead to merge conflicts which should be resolvable, but without pulling changes important bug fixes upstream will be left out. 
Finally regarding the branch choice, you're right, that's the correct usage, I'll update it.